### PR TITLE
Classifier: Revamp Image Toolbar with responsive styling

### PR DESCRIPTION
**Pull request by goplayoutside3**
~~Includes #4005.~~

## Package
lib-classifier

## Linked Issue and/or Talk Post
This is somewhat in preparation for building the "separate frames" view of multi-image subjects that use flipbook. The image toolbar is a consistent style and dimensions regardless of viewport size and layout, so this PR introduces responsiveness.

## Describe your changes

This PR primarily refactors styling of ImageToolbar's Button component as well as FieldGuideButton. I tried to unify the dimensions and padding of those two because they occupy the same `toolbar` grid area.

Other changes
    - Removed unneeded `span` elements from ZoomInButton and ZoomOutButton
    - Removed `defaultProps` from ImageToolbar components
    - Sized-up InvertIcon's svg by 3px to match the other icons
    - Miscellaneous linting

Tests and storybook changes:
    - Converted ImageToolbar > Button to RTL tests
    - Converted FieldGuideButton to RTL tests
    - Removed additional duplicate or not needed tests
    - Added a story for ImageToolbar's smaller styling

## How to Review

View any project that uses the Image Toolbar and resize your browser width to view how the styling responds to shifts in layout. Screenshot of comparison below. 

Desktop styling (macbook)
desktop

Smaller viewer styling (iphone 12 before & after)
old
new

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

**Comment by shaunanoordin** [comment]
I'm self-assigning this for a review, hope that's cool! I'll try to throw as much as I can on the UI testing.

**Review by shaunanoordin — COMMENTED** [review]
## PR Review

Package: lib-classifier
Affects: Image Toolbar on Subject Viewer

This PR is primarily an update to the Image Toolbar to give it responsive styling.

- New behaviour: the image toolbar should grow _narrower_ when the viewport is _narrower._ This prevents the image toolbar from taking up too much space when viewed on mobile.
- All other behaviour _should(?)_ be the same, except...
- ❓ _other_ new behaviour: on **tall images,** (i.e. vertical size > horizontal size) the ImageToolbar is **no longer sticky** (relative to the viewport).
  - I'm not certain if this is intended or not.
  - See the Dev Notes for full details.

Code read:
- most components have minor refactors that shouldn't change their behaviour
- most notable change is with DefaultLayout, where...
  - (1) the subject viewer's grid has a `clamp()` added - the second column (i.e. where the image toolbar is) _should_ have its width vary from 3rem to 4.5rem, depending on viewport width.
  - (2) The CSS between the styled `div` surrounding the toolbar (aka ImageToolbarContainer) and the ImageToolbar itself have been reshuffled. (EDIT: This may be pertinent; see the dev notes.)

### Dev Notes

Tested on localhost with macOS + Chrome 108, with [a test project full of tall images.](https://local.zooniverse.org:3000/projects/darkeshard/test-project-2022/classify/workflow/3581?env=staging)

The issue/the thing I'm not sure is intended new behaviour: on **tall images,** the ImageToolbar is **no longer sticky** 

Replication steps:
- Set browser viewport to a consistent wide size, like say 1000x480
- Open the classifier. Workflow's subjects should be nothing but tall images (preferably of dashing space aliens called Ultramen)
- Scroll down.
- Observe the position of the Image Toolbar.

_Video: on [master (frontend.preview)](https://frontend.preview.zooniverse.org/projects/darkeshard/test-project-2022/classify/workflow/3581?env=staging), the Image Toolbar "sticks" to the viewport. This means that when I'm viewing the feet of the Ultraman, I can still click on the zoom/rotate/etc buttons._

https://user-images.githubusercontent.com/13952701/208698068-2e987496-8a26-4871-8d1a-7cc6a38fd766.mov

_Video: on this branch (605970881da4778c63020fe69343704979e88e08), the Image Toolbar "sticks" for only a few pixels, but otherwise stays close to the top of the subject image (i.e. the Ultraman's head). So, when I scroll to view the Ultraman's feet, I can no longer view the Image Toolbar_

https://user-images.githubusercontent.com/13952701/208698659-821d0ebb-6e1b-46b8-ac7a-f4ac85383895.mov

**Analysis:**

To investigate the issue, I looked for the `div` with `position: sticky` then plonked a big red border to examine it.

_Screenshot (left): the sticky div on master. Screenshot (right): the sticky div on this branch._

Screenshot 2022-12-20 at 22 50 20 Screenshot 2022-12-20 at 22 50 13

OK, here I think I see that the "sticky" div IS still trying to be sticky... it's just that on this branch, the height of the sticky div is so _tall,_ it has _nowhere to move._

Looking back in the code, I _think(? citation needed?)_ I know where the problem is, and it's related to item (2) in the code read.

In master/prior to this PR, the DOM+CSS would look something like...

```
<StyledImageToolbarContainer>  // has grid-area=toolbar
  <StyledImageToolbar />  // has position=sticky
</StyledImageToolbarContainer>
```

Whereas in this PR, we have...

```
<ImageToolbarContainer>  // has grid-area=toolbar AND position=sticky
  <ImageToolbar />  // no additional CSS
</ImageToolbarContainer>
```

By moving position=sticky out of the ImageToolbar and into the container, the container now doesn't have _its own container_ where it has space to move around.

_Screenshot (left): master. Screenshot (right): this branch. In both cases, the div with position=sticky (marked red) can only move within the bounds of its containing div (marked blue)_

Screenshot 2022-12-20 at 23 32 29 Screenshot 2022-12-20 at 23 32 20

### Status

I'm not sure if the new "image toolbar isn't sticky for tall images" behaviour is intended or not, but otherwise, every other check looks good.

If that new behaviour _is_ intended, please let me know and I'll switch my review to an Approve.

If _not,_ then my recommendation is to revert the shuffling of the position=sticky between the ImageToolbar and the ImageToolbarContainer in DefaultLayout.

**Comment by eatyourgreens** [comment]
Transcription projects use tall subjects. Will this PR change how they behave? I’m a bit nervous about accidentally changing volunteers’ expectations for live projects.

File: packages/lib-classifier/src/components/Classifier/components/ImageToolbar/ImageToolbar.js
Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/ImageToolbar/ImageToolbar.js:18]: This allows you to pass the standard Grommet styling props to the toolbar. So different layouts can use different padding etc.

Why was it removed? Easy enough to restore, though.

**Comment by goplayoutside3** [comment]
Thanks @shaunanoordin for catching the sticky behavior! It's not an intended behavior change, so I've reverted the styled-components in DefaultLayout. The toolbar stickiness should behave the same on this branch and on production.

File: packages/lib-classifier/src/components/Classifier/components/ImageToolbar/ImageToolbar.js
Comment (goplayoutside3) [reply packages/lib-classifier/src/components/Classifier/components/ImageToolbar/ImageToolbar.js:18]: It was removed because as Shaun mentioned, I couldn’t find cases where props were explicitly passed to ImageToolbar either. However, when investigating sticky behavior, it turns that passing  generalized `…props` in the ImageToolbar component file is how the sticky css rules are applied to the toolbar `div` via DefaultLayout. I added a code comment in ImageToolbar as a reference to how the css `…props` are passed from the page layout.

**Review by seanmiller26 — APPROVED** [review]
After a chat and a walk through on local staging everything looks good and functions correctly. The icons do not shrink below suggested sizes, padding is correct, and the ratio of toolbar to subject size is a nice solution.

One question is whether this will be affected by the future breakpoints we will likely implement. Since this responsiveness is more fluid than breakpoints, just curious how that interplay will work.

**Comment by goplayoutside3** [comment]
@shaunanoordin would you like to review this PR again code-wise before I merge? Or good as is?

**Comment by shaunanoordin** [comment]
Let me run a quick manual re-test of this PR right now, but if the toolbar stickiness is reverted, I think everything should be OK! Be back in 10 mins with an updated review. 👌

**Review by shaunanoordin — APPROVED** [review]
## PR Review (Update)

Follows [earlier review](https://github.com/zooniverse/front-end-monorepo/pull/4008#pullrequestreview-1224606545)

The latest updates addresses the earlier issue of "on tall images, the ImageToolbar is no longer sticky", so this PR looks really good now!

### Testing

Tested on macOS + Chrome, running `yarn dev` on `app-project`

Testing steps:
- Selecting a variety of 1. viewport sizes and 2. subject dimensions (see Scenarios below)...
- ...we view the Classify page of various projects to confirm that the Image Toolbar continues to be functional to a user on the Classifier.
- Definition of "functional to a user": 1. the Image Toolbar should always be visible (sticky) and interact-able (not blocked by other elements) as the user scrolls down the page and/or otherwise views the Subject, 2. the buttons on the toolbar should be of reasonable interact-able size

**Scenario 1:** tall images
URL: https://local.zooniverse.org:3000/projects/darkeshard/test-project-2022/classify/workflow/3581
Viewport: tall & narrow (500x800)

image

**Scenario 2:** wide view
URL: https://local.zooniverse.org:3000/projects/darkeshard/test-project-2021/classify/workflow/20333?env=production
Viewport: short & wide (800 x 400)
 
image

**Scenario 3:** TESS
URL: https://www.zooniverse.org/projects/nora-dot-eisner/planet-hunters-tess/classify/workflow/11235?env=production
Viewport: standard (800x600)

image


### Status

LGTM! 👍

**Pull request merged**
